xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-api/src/main/java/org/xwiki/search/solr/internal/EmbeddedSolr.java

```diff
@@ -463,9 +470,26 @@ private void createSearchCore() throws IOException
     private void createCacheCore(Path corePath, String solrCoreName) throws IOException
     {
         // Indicate the path of the data
-        Path dataPath = getCacheCoreDataDir(corePath, solrCoreName);
-        FileUtils.write(corePath.resolve("core.properties").toFile(), "dataDir=" + dataPath, StandardCharsets.UTF_8,
-            true);
+        Path dataDir = getCacheCoreDataDir(corePath, solrCoreName);
+        File corePropertiesFile = corePath.resolve(CORE_PROPERTIES_FILENAME).toFile();
+        Properties coreProperties = new Properties();
+        // we used to append this property to this file.
+        // I'm not sure any other properties are ever in here, but jic let's be passive
+        // So I load the existing file if it exists, and add my property.
+        if(corePropertiesFile.exists()) 
+        {
+        	try(BufferedReader in = new BufferedReader(new FileReader(corePropertiesFile, StandardCharsets.UTF_8)) )
+        	{
+        		coreProperties.load(in);
+        	}
+        }
+        coreProperties.setProperty(DATA_DIR_PROPERTY, dataDir.toString());
+        // Normally we write using the Apache Commons FileUtils, but this is a properties file. 
```

Comment:
I don't think you need to mention the fact that FileUtils used to be used, using Properties tool is more connect and there is no reason to justify it (that's more something for the pull request explanation then the code).